ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs

```diff
@@ -132,23 +147,162 @@ class ( SL.ShelleyBasedEra era
 
       , ToCBOR (Core.Witnesses era)
 
+      , Eq (ShelleyApplyTxError era)
+      , FromCBOR (ShelleyApplyTxError era)
+      , Show (ShelleyApplyTxError era)
+      , ShowProxy (ShelleyApplyTxError era)
+      , ToCBOR (ShelleyApplyTxError era)
       ) => ShelleyBasedEra era where
```

Comment:
@nc6 Do the additions to this PR seem sane? The key idea is that -- at the Consensus level -- Alonzo will differ from the preceding eras in that its `applyTx` failures can additionally be due to "warnings promoted to failures". Think of it like the local node has set `-Werror`, where, in this case, the warnings would (eventually) be the Plutus error messages.

Reply:
Outdated -- PR input-output-hk/cardano-ledger-specs#2379 supplants this